partition_snapshot_row_cursor.hh

```diff
@@ -198,17 +268,41 @@ class partition_snapshot_row_cursor final {
     // When throws, the cursor is invalidated and its position is not changed.
     bool advance(bool keep) {
         memory::on_alloc_point();
-        position_in_version::less_compare heap_less(_schema);
+        version_heap_less_compare heap_less(*this);
         assert(iterators_valid());
         for (auto&& curr : _current_row) {
             if (!keep && curr.unique_owner) {
                 mutation_partition::rows_type::key_grabber kg(curr.it);
                 kg.release(current_deleter<rows_entry>());
+                if (_reversed && curr.it) [[unlikely]] {
+                    if (curr.rows->begin() == curr.it) {
+                        _background_continuity |= bool(curr.it->continuous());
+                        curr.it = {};
+                    } else {
+                        curr.continuous = curr.it->continuous();
+                        --curr.it;
+                    }
+                }
             } else {
-                ++curr.it;
+                if (_reversed) [[unlikely]] {
+                    if (curr.rows->begin() == curr.it) {
+                        _background_continuity |= bool(curr.it->continuous());
+                        curr.it = {};
+                    } else {
+                        curr.continuous = curr.it->continuous();
+                        --curr.it;
+                    }
+                } else {
+                    ++curr.it;
+                    if (curr.it) {
+                        curr.continuous = curr.it->continuous();
+                    }
+                }
             }
             if (curr.version_no == 0) {
-                _latest_it = curr.it;
+                if (curr.it) { // in _reversed case above the iterator may be cleared
+                    _latest_it = curr.it;
+                }
             }
```

Comment:
If you push this `if` into the `if (curr.it)` below, you won't need to check `if (curr.it)` twice.